### ISSUE TITLE
White Textures With Nvidia GTX 1080 (?)

### ISSUE THREAD

**Issue by Baker7** [DarkPlacesEngine/darkplaces#115]
I have a Zircon DarkPlaces Beta user Alforno (Windows user, Nvidia GTX 1080) who said he gets white textures only with Zircon effects level 5 and 6 (these are equivalent to Xonotic 2nd highest and Xonotic highest including real-time lights).

These are set at Level 5 and 6

r_shadow_realtime_dlight_shadows 1
r_shadow_realtime_world_lightmaps 1
r_shadow_shadowmapping 1 

His screenshot:

![alforno_e1m3_lightissue_settings_5_6](https://github.com/DarkPlacesEngine/darkplaces/assets/160253/6f7607f2-04ea-40f4-a522-92bbf5df8bf4)

I imagine it is lightmap texture upload somehow failing.  And I am using Nvidia and I suspect many Zircon users are and I've heard this problem from others, so I imagine it is a driver bug.

The lightmap process in DarkPlaces Beta, to the best of my knowledge, uses multiple threads.  I also didn't see any recent commits related to lightmaps.

I looked through recent DarkPlaces Beta commits to see if there were any driver related exceptions added.

Just sharing the information because I've noticed from code comments a heightened awareness of driver quirks.

**Comment by Baker7** [comment]
Alforno updated drivers and also has the same issue in a very early Zircon DarkPlaces Beta #2 (before I touched very much) and Zircon Alpha (based off Xonotic 0.8.2 DarkPlaces 2017) -- so whatever is up even affects old DarkPlaces.

White textures in ALL 3 when using feature, but does not have white texture look with regular settings.

/End supplying information.

**Comment by bones-was-here** [comment]
I'm guessing the screenshot is of Quake's `demo1.dem` which looks fine to me on nvidia with "FULL" RT lighting applied, followed by the cvars posted above.  It's too bright for my taste with those settings, but not broken.

Is there a way to reproduce this in current unmodified DarkPlaces?  If not, I'm doing to say this is not a DP bug.

**Comment by Baker7** [comment]
Do you have a link to a binary he could try to check?  I can't build a unmodified binary on Windows.

**Comment by bones-was-here** [comment]
Yes the xonotic.exe in the latest autobuild or engine-only zip at https://beta.xonotic.org/autobuild/ can be renamed or used with an argument such as `-quake` to run any supported game.

**Comment by Baker7** [comment]
For reference, I made a zip: https://www.moddb.com/mods/zircon-engine/downloads/test-darkplaces-settings-4-5 and provided it to alforno for him to check it out.

There is a darkplaces_setting4_xonotic_autobuild_20240109.bat and a darkplaces_setting5_xonotic_autobuild_20240109.bat.  

setting4.bat

-quake +cl_decals 1 +cl_decals_fadetime 4 +cl_decals_models 0 +cl_particles_quality 1.01 +gl_flashblend 0 +mod_q3bsp_nolightmaps 0 +r_bloom 1 +r_coronas_occlusionquery 1 +r_depthfirst 2 +r_drawdecals_drawdistance 500 +r_drawparticles_drawdistance 2000 +r_glsl_deluxemapping 1 +r_glsl_offsetmapping 0 +r_glsl_offsetmapping_reliefmapping 0 +r_motionblur 0.4 +r_shadow_gloss 1 +r_shadow_realtime_dlight 1 +r_shadow_realtime_dlight_shadows 0 +r_shadow_realtime_world 0 +r_shadow_realtime_world_shadows 0 +r_shadow_realtime_world_lightmaps 0 +r_shadow_shadowmapping 0 +r_shadow_usenormalmap 1 +r_showsurfaces 0 +r_sky 1 +r_subdivisions_tolerance 4 +r_water 1 +r_water_resolutionmultiplier 1

setting5.bat

-quake +cl_decals 1 +cl_decals_fadetime 10 +cl_decals_models 0 +cl_particles_quality 1.02 +gl_flashblend 0 +mod_q3bsp_nolightmaps 0 +r_bloom 1 +r_coronas_occlusionquery 1 +r_depthfirst 2 +r_drawdecals_drawdistance 500 +r_drawparticles_drawdistance 2000 +r_glsl_deluxemapping 1 +r_glsl_offsetmapping 1 +r_glsl_offsetmapping_reliefmapping 0 +r_motionblur 0.4 +r_shadow_gloss 1 +r_shadow_realtime_dlight 1 +r_shadow_realtime_dlight_shadows 1 +r_shadow_realtime_world 1 +r_shadow_realtime_world_shadows 0 +r_shadow_realtime_world_lightmaps 1 +r_shadow_shadowmapping 1 +r_shadow_usenormalmap 1 +r_showsurfaces 0 +r_sky 1 +r_subdivisions_tolerance 2 +r_water 1 +r_water_resolutionmultiplier 1 +r_shadow_gloss 1

**Comment by Baker7** [comment]
Using the DarkPlaces binary you linked .. alforno tried it and same issue ...

"4 worked and 5 had the white texture issue."

![darkplaces_still_white](https://github.com/DarkPlacesEngine/darkplaces/assets/160253/4f48e5dd-9ca4-4307-bcfe-22094fbec7b1)

**Comment by bones-was-here** [comment]
What are the steps to reproduce (config and/or assets and/or gamecode) on a graphics card other than alforno's specific one?

**Comment by Baker7** [comment]
>What are the steps to reproduce (config and/or assets and/or gamecode) on a graphics card >other than alforno's specific one?

No evidence that any other graphics card has issue.

I don't experience the issue (I use nvidia but not a 1080) and I have not had anyone else other than Alforno report this as an issue.

To try to get the problem, all you have to do is download  https://www.moddb.com/mods/zircon-engine/downloads/test-darkplaces-settings-4-5, unpack to Quake folder and double-click the settings5.bat --> done

**Comment by Baker7** [comment]
Update: alforno did some quality testing and says

 r_shadow_realtime_world 1 or 0 is the difference for him with the white textures for his GTX 1080
